src/Socket.php

```diff
@@ -36,16 +36,12 @@ public function __construct(LoopInterface $loop, $socket, Buffer $buffer = null)
 
     public function getLocalAddress()
     {
-        if ($this->socket !== false) {
-            return stream_socket_get_name($this->socket, false);
-        }
+        return $this->sanitizeAddress(@stream_socket_get_name($this->socket, false));
```

Comment:
This is all covered by the test suite and we now make sure to either return the correct address string or `null` if it's unknown. Here's how this works under the hood:

```php
>>> $f = stream_socket_client('google.com:80');
=> stream resource #279
>>> stream_socket_get_name($f, true);
=> "216.58.205.238:80"
>>> fclose($f);
=> true
>>> @stream_socket_get_name($f, true);
PHP warning:  stream_socket_get_name(): supplied resource is not a valid stream resource on line 1
=> false
>>> @stream_socket_get_name(false, true);
PHP warning:  stream_socket_get_name() expects parameter 1 to be resource, boolean given on line 1
=> false
```

This uses a client socket for simplicity, the server socket emits similar errors but has far more error situations, e.g. the server socket does not have a remote address and returns `false` without emitting an error (unless you're on HHVM).